R/execution.r

```diff
@@ -92,12 +92,20 @@ execute = function(request) {
     handle_message = identity
     handle_warning = identity
   } else {
-    handle_value = function (obj) {
-        data = list()
-        data['text/plain'] = paste(capture.output(print(obj)), collapse="\n")
-        send_response("execute_result", request, 'iopub',
-                  list(data=data, metadata=namedlist(),
-                  execution_count=execution_count))
+    handle_value <- function (obj) {
+        data <- list()
+        if (getOption('jupyter.rich_display')) {
+            for (mime in getOption('jupyter.result_mimetypes')) {
+                r <- mime2repr[[mime]](obj)
+                if (!is.null(r)) data[[mime]] <- r
+            }
+        } else {
+            data[['text/plain']] <- repr_text(obj)
```

Comment:
For my own curiosity, what's the difference between setting `data['text/plain']` and `data[['text/plain']]`? In a quick test, the result appears to be the same, but I seem to remember they're not equivalent.